tools/src/main/java/org/apache/kafka/trogdor/coordinator/TaskManager.java

```diff
@@ -395,11 +401,13 @@ public Void call() throws Exception {
                 log.error("Unable to find nodes for task {}", task.id, e);
                 task.doneMs = time.milliseconds();
                 task.state = TaskStateType.DONE;
+                trogdorMetrics.recordDoneTask();
```

Comment:
It would be better to create a method in ManagedTask like `setState(...)` and increment the appropriate metric if the state had changed.  Then we could move all these `state = foo` lines over to using that API.  This would prevent making a mistake here.